source/core/coverage/ut_coverage.pks

```diff
@@ -0,0 +1,54 @@
+create or replace package ut_coverage authid current_user is
+
+  -- total run coverage information
+  subtype t_full_name   is varchar2(500);
+  subtype t_object_name is varchar2(250);
+
+  subtype t_line_executions is binary_integer;
+  -- line coverage information indexed by line no.
+  type tt_lines is table of t_line_executions index by binary_integer;
+  --unit coverage information record
+  type t_unit_coverage is record (
+    covered_lines   binary_integer := 0,
+    uncovered_lines binary_integer := 0,
+    total_lines     binary_integer := 0,
+    executions      number(38,0) := 0,
+    lines           tt_lines
+  );
+
+  -- coverage information indexed by full object name (schema.object)
+  type tt_program_units is table of t_unit_coverage index by t_full_name;
+
+  -- total run coverage information
+  type t_coverage is record(
+    covered_lines   binary_integer := 0,
+    uncovered_lines binary_integer := 0,
+    total_lines     binary_integer := 0,
+    executions      number(38,0)   := 0,
+    objects         tt_program_units
+  );
+
+  function  get_coverage_id return integer;
+
+  function  coverage_start(a_schema_names ut_varchar2_list := ut_varchar2_list(sys_context('userenv','current_schema'))) return integer;
```

Comment:
This might have been true on Oracle 10. With 11g XE it works fine.
I've tested the provided example and go the following:
```sql
conn ut3/ut3
create or replace procedure definer_proc is
  begin
    dbms_output.PUT_LINE('Current user=' || sys_context('userenv', 'current_user'));
    dbms_output.PUT_LINE('Session user=' || sys_context('userenv', 'session_user'));
    dbms_output.PUT_LINE('Current schema=' || sys_context('userenv', 'current_schema'));
  end definer_proc;
/
create or replace procedure invoker_proc authid current_user is
  begin
    dbms_output.PUT_LINE('Current user=' || sys_context('userenv', 'current_user'));
    dbms_output.PUT_LINE('Session user=' || sys_context('userenv', 'session_user'));
    dbms_output.PUT_LINE('Current schema=' || sys_context('userenv', 'current_schema'));
  end invoker_proc;
/
grant execute on definer_proc to ut3_user;
grant execute on invoker_proc to ut3_user;
begin ut3.definer_proc; end;
/
begin ut3.invoker_proc; end;
/

sql> begin ut3.definer_proc; end;
Current user=UT3
Session user=UT3
Current schema=UT3
sql> begin ut3.invoker_proc; end;
Current user=UT3
Session user=UT3
Current schema=UT3
```

```sql
conn ut3_user/ut3_user
begin ut3.definer_proc; end;
/
begin ut3.invoker_proc; end;
/  
sql> begin ut3.definer_proc; end;
Current user=UT3
Session user=UT3_USER
Current schema=UT3
sql> begin ut3.invoker_proc; end;
Current user=UT3_USER
Session user=UT3_USER
Current schema=UT3_USER

```